File: apps/engine_sdk/lib/evil_engine/plugin/data_store_adapter.ex
defmodule EvilEngine.Plugin.DataStoreAdapter do
  @moduledoc """
  External data store integration for BPMN DataStore references.

  **Not implemented in v1.** Registration is accepted and ignored at
  runtime. BPMN DataStores are a parser no-op in this release. Do not
  depend on this behaviour being invoked.

  Unique per `store_id` if this capability is ever wired.
  """

  @callback store_id() :: String.t()
  @callback read(key :: String.t(), opts :: keyword()) :: {:ok, term()} | {:error, term()}
  @callback write(key :: String.t(), value :: term(), opts :: keyword()) :: :ok | {:error, term()}
end

File: packages/js/sdk/src/plugin/data-store-adapter.ts
/**
 * Handler interface for data store adapter plugins.
 *
 * Not implemented in v1. `registerDataStoreAdapter` is accepted and unused at
 * runtime — BPMN DataStores remain a parser no-op.
 */
export interface DataStoreAdapterHandler {
  storeId(): string;
  read(key: string, options?: Record<string, unknown>): Promise<unknown>;
  write(key: string, value: unknown, options?: Record<string, unknown>): Promise<void>;
}
